dashboard/src/get_elements.py

```diff
@@ -34,44 +37,61 @@
             You can specify property names to ONLY includes in the includes list
             This is working with XOR logic, if there are elements in includes, excludes will not be considered
 '''
+
+def smart_cast(value):
+    tests = [float, str, bool]
+    if value in ['True', 'False']:
+        return bool(value)
+    if value.count('.') > 1 or value.upper().isupper():
+        return str(value)
+    for test in tests:
+        try:
+            return test(value)
+        except ValueError:
+            continue
+    return value
+
 class GetElements:
     def __init__(self, dataset=None, ifc_type=None, debug='nodebug', properties_options={'excludes': [], 'includes': []}, **kwargs):
         self.ifc_type = ifc_type
         self.debug = debug
         self.properties_options = properties_options
-        self.elements = {}
+        self.elements = []
         self.flat_elements = {}
         self.properties = {}
         self.properties['name'] = []
         self.properties['pset'] = []
+        self.host = 'staging'
+        self.types_ref = {}
         if dataset is not None:
             self.access_token = dataset['access_token'][0]
             self.cloud_pk = str(dataset['cloud_id'][0])
             self.project_pk = str(dataset['project_id'][0])
-            self.ifc_pk = str(dataset['ifc_id'][0])
+            self.ifc_pks = str(dataset['ifc_id'][0]).split(',')
+            self.host = str(dataset['host'][0])
         else:
             from dotenv import load_dotenv
-            load_dotenv()
+            load_dotenv('.env')
             import os
-            with open('.env', 'r') as env:
-                self.access_token = os.getenv('TOKEN')
-                self.cloud_pk = os.getenv('CLOUD_ID')
-                self.project_pk = os.getenv('PROJECT_ID')
-                self.ifc_pk = os.getenv('IFC_ID')
+            self.access_token = os.getenv('TOKEN')
+            self.cloud_pk = os.getenv('CLOUD_ID')
+            self.project_pk = os.getenv('PROJECT_ID')
+            self.ifc_pks = os.getenv('IFC_ID').split(',')
+            print(self.ifc_pks)
+            self.host = os.getenv('HOST')
 
     def config(self):
         configuration = bimdata_api_client.Configuration()
-        configuration.api_key['Authorization'] = self.access_token
-        configuration.api_key_prefix['Authorization'] = 'Bearer'
-        configuration.host = 'https://api-staging.bimdata.io'
+        configuration.access_token = self.access_token
+        configuration.host = f'https://api-{self.host}.bimdata.io'

```

Comment:
host should be the API url directly. We'll remove the '-beta' in the prod url soon

Reply:
done